test/jest_integration/saml_auth.test.ts

```diff
@@ -326,12 +326,14 @@ describe('start OpenSearch Dashboards server', () => {
 
     await driver.wait(until.elementsLocated(By.xpath(tenantNameLabelXPath)), 10000);
 
-    const tenantName = await driver.findElement(By.xpath(tenantNameLabelXPath)).getText();
+    const localStorageItem = await driver.executeScript(
```

Comment:
Know why the text wasn't loading?  These test are suppose to make sure the UX is rendering the correct content, and this change effectively works around inspecting the UI - this is a major hit to the functionality of the test to use the localstorage state.

Reply:
Hi @peternied, thanks for the review! I totally agree that we should test out the actual UI elements. However, according to the experiment @cwperks and I did on my local machine, it is not about the loading of that UI elements. It is something about the selenium webdriver, and I have also tried out different webdrivers such as chromedriver and they all have a very high chance for not go through the second `window.reload()` for updating the saved tenant according to the localstorage key. So there is nothing we can do about the webdriver for now :(

However, I'm willing to figure out a work around as an another approach if it is necessary.

Reply:
@RyanL1997 I was just looking at the requests that happen after login and I believe what we can do is wait for a response from `http://localhost:5601/api/v1/multitenancy/tenant` before clicking on the user dropdown and viewing the tenant info. This endpoint is called after `login` to get the current tenant info from what I can see.

Reply:
With playwright its possible to do a `page.waitForResponse(urlOrPredicate... `, but I'm not sure if selenium has something similar.

https://playwright.dev/docs/api/class-page#page-wait-for-response

Reply:
Maybe we can dig into this a little more to see what the test should do verses what is hard to do,  Quickly reading the test:
1. Go to the overview page
2. Sign in (Not as clear how this is actually done in the test...)
3. Select the global tenant, confirm
4. Sign out
5. Sign back in
6. Skip the popup-ux
7. Wait until the tenant label appears and show `Global`

Seems like knowing we are on the global tenant is a critical part of the test

Can we find an element that is blank until the tenant query has resolved?  We could add an element with an ID that would be easy to find and wait for it to disappear when the tenant has loaded if that is part of the problem.

Reply:
@peternied Thanks for this info. The problem we are having right now is not about the loading of the UI. We are can load and locate this UI. However, it just doesn't switch to the expected tenant (It sometimes stays as "private" [1]). For now, I am trying to implement a retry for this scenario, and I am also taking a look into @cwperks 's suggestion to see how much effort will it take us to switch to another test frame work.

Reference:
[1]: 
<img width="797" alt="Screenshot 2022-12-05 at 1 47 07 PM" src="https://user-images.githubusercontent.com/109499885/205717934-3bbb1a72-7ec6-4e69-abba-c69be40c92e5.png">

Reply:
> However, it just doesn't switch to the expected tenant (It sometimes stays as "private" [1])

This sounds like a product bug?

Reply:
I don think so, because, based on the experiment I did so far, it wont happen to the regular browser, and it is just the webdriver we are using got a high chance to not reload the window for updating tenant information according to the local storage.

Reply:
@RyanL1997 Another thought, maybe its possible to use `until.elementTextIs(element, text)`?

https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/until.html